UM/Math/Polygon.py

```diff
@@ -175,8 +175,10 @@ def intersectsPolygon(self, other: "Polygon") -> Optional[Tuple[float, float]]:
 
         polygon_me = ShapelyUtil.polygon2ShapelyPolygon(self)
         polygon_other = ShapelyUtil.polygon2ShapelyPolygon(other)
-
-        if not (polygon_me.is_valid and polygon_other.is_valid):
+        if (not polygon_other._is_empty and not polygon_me._is_empty):
```

Comment:
```suggestion
        if not polygon_other.is_empty and not polygon_me.is_empty:
```
There is also a public variant so that one should be used.